conda-store-server/conda_store_server/_internal/server/app.py

```diff
@@ -293,15 +293,32 @@ async def exception_handler(request, exc):
             # docker registry api specification does not support a url_prefix
             app.include_router(views.router_registry)
 
-        if self.enable_ui:
+        if self.enable_metrics:
             app.include_router(
-                views.router_ui,
-                prefix=trim_slash(self.url_prefix) + "/admin",
+                views.router_metrics,
+                prefix=trim_slash(self.url_prefix),
             )
 
+        if self.additional_routes:
+            for path, method, func in self.additional_routes:
+                getattr(app, method)(path, name=func.__name__)(func)
+
+        if isinstance(self.conda_store.storage, storage.LocalStorage):
+            self.conda_store.storage.storage_url = (
+                f"{trim_slash(self.url_prefix)}/storage"
+            )
+            app.mount(
+                self.conda_store.storage.storage_url,
+                StaticFiles(directory=self.conda_store.storage.storage_path),
+                name="static-storage",
+            )
+
+        # This needs to come at the end because if the UI is enabled,
+        # it becomes the catch all route
+        if self.enable_ui:
```

Comment:
The diff view here is confusing. What I did was move this entire if-block (`if self.enable_ui`) down to the bottom of this function (`init_fastapi_app`). And within the if-block, I moved the UI router from near the top to the very end.

Why does this need to be moved around? Because the order in which the routes are defined matters.

For example, if I put the `/favicon.ico` definition after the UI app, then a request for `/favicon.ico` will end up loading the UI app instead of the favicon.